pylint/pyreverse/utils.py

```diff
@@ -189,11 +187,11 @@ def leave(self, node):
 class LocalsVisitor(ASTWalker):
     """Visit a project by traversing the locals dictionary."""
 
-    def __init__(self):
+    def __init__(self) -> None:
         super().__init__(self)
-        self._visited = set()
+        self._visited: set[nodes.NodeNG] = set()
 
-    def visit(self, node):
+    def visit(self, node: nodes.NodeNG) -> Any:
```

Comment:
Is this truly `Any`? Don't all `visit_` methods return `None`?

Reply:
With the updated typing, should this be `Union[Tuple[ClassDiagram], Tuple[PackageDiagram, ClassDiagram], None]`?

If that's the case, the visit method is probably not type compatible with the base implementation in `ASTWalker.visit` with is annotated to only return `None`. _The type of the child implementation should be a subclass of the parent impl._

Reply:
Good catch. I can update both methods to use `Union[Tuple[ClassDiagram], Tuple[PackageDiagram, ClassDiagram], None]`.

Reply:
On second thought: 
This also comes down to design problems.
`ASTWalker.visit` does in fact always return `None` (there is no `return` inside the method).
`LocalsVisitor` (a subclass of `ASTWalker`) overrides `visit` and returns whatever the `leave` method for this node type returns. That's why I put `Any` here in the first place.

Currently the `ASTWalker` class is never used on its own. We could simply remove it and move the relevant methods into `LocalsVisitor`.

Reply:
That should work. Alternatively, `ASTWalker.visit` can be annotated with the return type from `LocalsVisitor.visit`. Doesn't matter that only `None` is actually returned there. But with that the subclass implementation is compatible.